src/charm.py

```diff
@@ -301,8 +257,16 @@ def _requested_receivers(self) -> Tuple[ReceiverProtocol, ...]:
         requested_receivers = requested_protocols.intersection(set(self.tempo.receiver_ports))
         return tuple(requested_receivers)
 
+    @property
+    def trace_retention_period_hours(self) -> int:
+        """Trace retention period for the compactor."""
+        # if unset, default to 30 days
+        trace_retention_period_hours = cast(int, self.config.get("retention_period_hours", 720))
```

Comment:
...unless you adopt the suggestion elsewhere of making this a string input.  Then this should stay

Reply:
umh, "unless you omit a default, the user cannot leave it unset"? are you sure? I'd have said that if you do have a default value, as we do, the user can leave it unset and it will fall back to the default value. so we're good?
Am I getting confused by the double negation or are we not not saying the same thing?

Reply:
sorry probably a bad sentence :D 

What I mean is that, if an integer config value starts with a default value, I don't think its possible for the user to "unset" it so the default in `.get()` will never be used (or if it is used, something else had an error)

Reply:
ah ok I get what you're saying